pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -239,6 +239,11 @@ func (cb *ClusterBuilder) buildSubsetCluster(opts buildClusterOpts, destRule *co
 func (cb *ClusterBuilder) applyDestinationRule(mc *MutableCluster, clusterMode ClusterMode, service *model.Service,
 	port *model.Port, proxyNetworkView map[network.ID]bool, destRule *config.Config, serviceAccounts []string) []*cluster.Cluster {
 	destinationRule := CastDestinationRule(destRule)
+
+	host := ""
+	if destinationRule != nil {
+		host = service.Hostname.String()
```

Comment:
I think if user had this config today (without the explicit SAN/caCerts), they would be broken for `*.google.com` since its exact match though?

Exact match on `*.x.y` doesn't make sense at all. IMO we can either not automatically set the SAN check for wildcard (as short term action), and long term allow prefix. 

We could go directly to long term, just don't want to scope creep this more...

Reply:
That's right. I thought you meant applying an SE with a wildcard is broken (cannot be applied). I didn't catch the example above did not have `resolution: DNS`. 
I think we will go with not setting SAN when wildcard is set and work toward auto_san/auto_sni for long term